store/tikv/client.go

```diff
@@ -129,8 +178,16 @@ func (c *batchCommandsClient) batchRecvLoop(cfg config.TiKVClient) {
 	}()
 
 	for {
-		// When `conn.Close()` is called, `client.Recv()` will return an error.
-		resp, err := c.client.Recv()
+		var err error = nil
+		var resp *tikvpb.BatchCommandsResponse = nil
+		failpoint.Inject("gotErrorInRecvLoop", func(_ failpoint.Value) {
+			err = errors.New("injected error in batchRecvLoop")
+		})
+		if err == nil {
+			// When `conn.Close()` is called, `client.Recv()` will return an error.
+			resp, err = c.client.Recv()
```

Comment:
This is not an effective way because of checking `err != nil` every time.
```go
		resp, err := c.client.Recv()
		failpoint.Inject("gotErrorInRecvLoop", func(_ failpoint.Value) {
			err = errors.New("injected error in batchRecvLoop")
			resp = nil
		})
```
Maybe better.

Reply:
They are different. `Recv` is a block call. I hope to trigger the fail point before call that function.